pkg/apis/core/v1/conversion_test.go

```diff
@@ -419,3 +419,65 @@ func Test_v1_PodStatus_to_core_PodStatus(t *testing.T) {
 		t.Errorf("Convert v1.PodStatus to core.PodStatus failed. expected corePodStatus.PodIPs[0].ip=%v found %v", v1PodStatus.PodIP, corePodStatus.PodIPs[0].IP)
 	}
 }
+
+func Test_core_NodeSpec_to_v1_NodeSpec(t *testing.T) {
+	// core to v1
+	var v1NodeSpec v1.NodeSpec
+	coreNodeSpec := core.NodeSpec{
+		PodCIDRs: []string{"10.0.0.0/24", "10.0.1.0/24"},
+	}
+	// convert
+	if err := corev1.Convert_core_NodeSpec_To_v1_NodeSpec(&coreNodeSpec, &v1NodeSpec, nil); nil != err {
+		t.Errorf("Convert core.NodeSpec to v1.NodeSpec failed with error %v", err.Error())
+	}
+
+	// validate results
+	if v1NodeSpec.PodCIDR != coreNodeSpec.PodCIDRs[0] {
+		t.Errorf("Convert core.NodeSpec to v1.NodeSpec failed. Expected v1.PodCIDR=%v but found %v", coreNodeSpec.PodCIDRs[0], v1NodeSpec.PodCIDR)
+	}
+
+	// match v1.PodIPs to core.PodIPs
+	for idx, _ := range coreNodeSpec.PodCIDRs {
+		if v1NodeSpec.PodCIDRs[idx] != coreNodeSpec.PodCIDRs[idx] {
+			t.Errorf("Convert core.NodeSpec to v1.NodeSpec failed. Expected v1.NodeSpec[%v]=%v but found %v", idx, coreNodeSpec.PodCIDRs[idx], v1NodeSpec.PodCIDRs[idx])
+		}
+	}
+}
+
+func Test_v1_NodeSpec_to_core_NodeSpec(t *testing.T) {
+	var coreNodeSpec core.NodeSpec
+	// Error case PodIP != PodIPs[0]
+	v1NodeSpec := v1.NodeSpec{
```

Comment:
As with podIPs - table-driven tests are much easier to think about.

Reply:
083bf4e resolves this..